draco/analysis/transform.py

```diff
@@ -178,94 +178,60 @@ def process(self, ss):
         sp : SiderealStream
             Dataset containing only the required products.
         """
-
-        # Define two functions that are used below
-        def find_key(key_list, key):
-            try:
-                return [tuple(x) for x in key_list].index(tuple(key))
-            except TypeError:
-                return list(key_list).index(key)
-            except ValueError:
-                return None
-
-        match_sn = True
-        if "stack" in ss.index_map:
-            match_sn = ss.index_map["stack"].size == ss.index_map["prod"].size
-
         # For each input in the file, find the corresponding index in the telescope instance
-        ss_keys = ss.index_map["input"][:]
+        ss_keys = ss.input
```

Comment:
@ssiegelx I think this might be a little better if I just add the `input_index` property onto `draco.core.telescope.TransitTelescope` to guarantee that it's present. That would allow removal of the `try... except` block. Thoughts?

It might also be worth implementing a `tools.find_inputs` function that will also figure out what field to match on. Let me know what you think. I should be able to do the implementation.

Reply:
Yeah this block of code for matching inputs is repeated in at least three places: here, `CollateProducts`, and the new method `tools.redefine_stack_index_map`.  Moving it to a single `tools.find_inputs` method is probably a good idea.